docs/merkletree/treetex/main.go

```diff
@@ -185,16 +192,20 @@ func drawLeaf(prefix string, index uint64) {
 // Returns a func to be called to close the node.
 func openInnerNode(prefix string, id compact.NodeID) func() {
 	attr := nInfo[id].String()
-	fmt.Printf("%s [%d.%d, %s, tier=%d\n", prefix, id.Level, id.Index, attr, id.Index-1)
+	fmt.Printf("%s [%d.%d, %s, tier=%d\n", prefix, id.Level, id.Index, attr, id.Level)
 	return func() { fmt.Printf("%s ]\n", prefix) }
 }
 
 // perfectInner renders the nodes of a perfect internal subtree.
-func perfectInner(prefix string, level uint, index uint64, top bool) {
+func perfectInner(prefix string, level uint, index uint64, top bool, leafData leafDataFunc) {
 	id := compact.NewNodeID(level, index)
 	modifyNodeInfo(id, func(n *nodeInfo) {
 		n.leaf = id.Level == 0
 		n.perfectRoot = top
+		if id.Level == 0 {
+			n.leaf = true
```

Comment:
This line is redundant (see 3 lines above).

Reply:
I meant to remove the one above - dinged it now.